pkg/yang/types_builtin.go

```diff
@@ -794,3 +930,17 @@ func (r YangRange) Contains(s YangRange) bool {
 	}
 	return true
 }
+
+// Frac returns the fractional part of f.
+func Frac(f float64) float64 {
+	return f - math.Trunc(f)
+}
+
+// pow10 returns 10^e without checking for overflow.
+func pow10(e uint8) uint64 {
+	var out uint64 = 1
+	for i := uint8(0); i < e; i++ {
+		out *= 10
+	} 
+	return out
+}
```

Comment:
Neat idea but for the sake of compactness I think I'll keep it as is, since performance is not critical here.